resilience4j2/src/main/java/com/linecorp/armeria/resilience4j/circuitbreaker/Resilience4jCircuitBreakerFactory.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.resilience4j.circuitbreaker;
+
+import com.linecorp.armeria.client.circuitbreaker.CircuitBreakerClientHandler;
+import com.linecorp.armeria.common.annotation.Nullable;
+
+import io.github.resilience4j.circuitbreaker.CircuitBreaker;
+import io.github.resilience4j.circuitbreaker.CircuitBreakerRegistry;
+
+/**
+ * A functional interface that represents a mapper factory, mapping a combination of host, method and path
+ * to a {@link CircuitBreaker} using a {@link CircuitBreakerRegistry}.
+ */
+public interface Resilience4jCircuitBreakerFactory {
```

Comment:
I assumed that since I added the `@UnstableApi` annotation to `package-info.java` there is no need to add to individual files. Let me know if my understanding is incorrect 🙏

Reply:
> I added the @UnstableApi annotation to package-info.java

Make sense for `@UnstableApi`. But I think we still need `@FunctionInterface`.

Reply:
Github highlight got me confused 😅   Done!

Reply:
I see, in that case I guess adding the annotation to `package-info` doesn't have a real purpose.
There aren't many public APIs in this module, so let me just copy-paste to relevant classes 😄

Reply:
Thanks for the change. 🙇